features/accounts/edit/profile-form/profile-form.tsx

```diff
@@ -1,41 +1,39 @@
-import React from 'react';
+import React, { FC } from 'react';
 import { Controller, useForm } from 'react-hook-form';
+import { DateObject } from 'react-multi-date-picker';
 
+import { DatePickerContainer } from '@/components/datePicker/datePickerContainer';
+import { useProfileFormSchema } from '@/features/accounts/edit/profile-form/use-profile-form-schema';
+import { Button } from '@/shared/ui';
 import { SelectBox } from '@/shared/ui/SelectBox';
 import { TextField } from '@/shared/ui/textField';
+import { DevTool } from '@hookform/devtools';
 import { zodResolver } from '@hookform/resolvers/zod';
 import { z } from 'zod';
 
 import style from './profile-form.module.scss';
 
-const schema = z.object({
-  aboutme: z.string(),
-  city: z.string(),
-  country: z.string(),
-  dateofbirth: z.date(),
-  firstname: z.string(),
-  lastname: z.string(),
-  username: z.string(),
-  // .min(6, { message: 'Minimum number of characters 6' })
-  // .max(30, { message: 'Maximum number of characters 30' })
-  // .regex(/^[0-9A-Za-z_-]+$/),
-});
-
-type FormValuesType = z.input<typeof schema>;
-
-export const ProfileForm = () => {
-  const {
-    clearErrors,
-    control,
-    formState: { errors },
-    handleSubmit,
-    setError,
-  } = useForm<FormValuesType>({
-    mode: 'onBlur',
+type FormValues = z.infer<ReturnType<typeof useProfileFormSchema>>;
+
+export const ProfileForm: FC = () => {
+  const schema = useProfileFormSchema();
+
+  const { control, handleSubmit } = useForm<FormValues>({
+    defaultValues: {
+      aboutMe: '',
+      birthDate: new DateObject().valueOf(),
```

Comment:
The value of the `birthDate` field cannot be represented by an instance of the `DateObject` class, because `useForm`'s `formState` cannot properly see a change in the value of such a data type and reflect it in the state of the `isDirty` field. The `DateObject` object must be explicitly cast to `string` or `number` (even though it is implicitly cast to a number, this still doesn't work).